core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -1708,6 +1723,10 @@ pub static mp_module_trezorui2: Module = obj_module! {
     /// ) -> CANCELLED:
     ///     """Homescreen used for indicating coinjoin in progress."""
     Qstr::MP_QSTR_show_busyscreen => obj_fn_kw!(0, new_show_busyscreen).as_obj(),
+
+    /// def show_welcome_screen() -> object:
+    ///     """Show logo icon with the model name at the bottom."""
+    Qstr::MP_QSTR_show_welcome_screen => obj_fn_kw!(0, new_show_welcome_screen).as_obj(),
```

Comment:
istm this does not need to be `obj_fn_kw`, can be `obj_fn_0` and not take arguments?

Reply:
Would be certainly nice, but now we do not have `obj_fn_0!` and even though I see the definitions of `obj_fn_1` and `obj_fn_2`, I do not know how to create it.

Tried

```rs
macro_rules! obj_fn_0 {
    ($f:expr) => {{
        #[allow(unused_unsafe)]
        unsafe {
            use $crate::micropython::ffi;

            ffi::mp_obj_fun_builtin_fixed_t {
                base: ffi::mp_obj_base_t {
                    type_: &ffi::mp_type_fun_builtin_0,
                },
                fun: ffi::_mp_obj_fun_builtin_fixed_t__bindgen_ty_1 { _0: Some($f) },
            }
        }
    }};
}
``` 

with receiving 
```
error[E0308]: mismatched types
    --> src/ui/model_tt/layout.rs:1729:52
     |
1729 |     Qstr::MP_QSTR_show_welcome_screen => obj_fn_0!(draw_welcome_screen).as_obj(),
     |                                                    ^^^^^^^^^^^^^^^^^^^ incorrect number of function parameters
     |
    ::: src/micropython/macros.rs:12:75
     |
12   |                 fun: ffi::_mp_obj_fun_builtin_fixed_t__bindgen_ty_1 { _0: Some($f) },
     |                                                                           ---- arguments to this enum variant are incorrect
```

Reply:
here it is: https://github.com/trezor/trezor-firmware/pull/2521/files#diff-9cf76d30275f61c11d21878407648fcfeb12737698c9fb0faade3209ea0014cd

Reply:
oh, that's exactly the same thing you have.

well the error seems to tell you that there's a wrong number of function parameters, indicating that you didn't remove parameters from `draw_welcome_screen` maybe?